src/xform_to_datalog/arcs_manifest_tree/particle_spec.h

```diff
@@ -17,9 +34,9 @@ namespace raksha::xform_to_datalog::arcs_manifest_tree {
 // This struct is used to return the members of a ParticleSpec after being
 // instantiated.
 struct InstantiatedParticleSpecFacts {
-  std::vector<raksha::ir::TagClaim> tag_claims;
-  std::vector<raksha::ir::TagCheck> checks;
-  std::vector<raksha::ir::Edge> edges;
+  std::vector<raksha::ir::InstanceFact<raksha::ir::TagClaim>> tag_claims;
```

Comment:
Do we need to have the `InstanceFact`? What is the advantage? Isn't the instantiated class representable using `TagClaim` itself?

Reply:
Yes, but doing so would require copying the `TagClaim`, which already contains a variable-sized `AccessPath` and will soon contain a `unique_ptr<Predicate>`.